pulpcore/cli/ansible/remote.py

```diff
@@ -0,0 +1,232 @@
+import click
+
+from typing import Any, Dict, Optional, cast, Callable, TypeVar
+
+from pulpcore.cli.common.generic import (
+    list_entities,
+    show_by_name,
+    destroy_by_name,
+)
+from pulpcore.cli.common.context import (
+    PulpContext,
+    pass_pulp_context,
+    pass_entity_context,
+    PulpEntityContext,
+)
+from pulpcore.cli.ansible.context import (
+    PulpAnsibleRoleRemoteContext,
+    PulpAnsibleCollectionRemoteContext,
+)
+
+RT = TypeVar("RT")  # return type
+
+
+def _type_callback(ctx: click.Context, param: Any, value: str) -> str:
+    group: click.Group = cast(click.Group, ctx.command)
+    if value == "role":
+        group.commands.pop("create_collection", None)
+        group.commands.pop("update_collection", None)
+    else:
+        group.commands.pop("create_role", None)
+        group.commands.pop("update_role", None)
+    return value
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "remote_type",
+    type=click.Choice(["role", "collection"], case_sensitive=False),
+    default="role",
```

Comment:
is "role" a good choice for the default?
I'd expect the whole ansible infrastructure to move towards collections (they can contain roles and are therefore much more generic).
cc @bmbouter @fao89

Reply:
Yea, collection is better. I think we want to encourage users to start using collections.

Reply:
+1 to collection

Reply:
+1 to collection